### PR TITLE
Fix NullReferenceException in DesignerAttribute.GetHashCode

### PR DESCRIPTION
**Pull request by hughbe**
And simplify the tests for other places where we threw NRE in GetHashCode/Equals

### CODE REVIEW

File: src/System.ComponentModel.TypeConverter/tests/Design/MenuCommandTests.cs [hunk 1]
```diff
@@ -98,6 +98,12 @@ public static IEnumerable<object[]> ToString_TestData()
                 },
                 "00000000-0000-0000-0000-000000000000 : 0 : Supported|Enabled|Visible|Checked"
             };
+
+            // .NET Framework throws NullReferenceException.
+            if (!PlatformDetection.IsFullFramework)
+            {
+                yield return new object[] { new MenuCommand(new EventHandler(EventHandler), null), " : Supported|Enabled|Visible" };
+            }
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/tests/Design/MenuCommandTests.cs]: Do these tests run on netfx?
cc: @ViktorHofer

File: src/System.ComponentModel.TypeConverter/tests/Design/Serialization/DesignerSerializerAttributeTests.cs [hunk 2]
```diff
@@ -51,28 +52,25 @@ public void Ctor_SerializerTypeName_BaseSerializerType(string serializerTypeName
...
+        public static IEnumerable<object[]> TypeId_TestData()
+        {
+            yield return new object[] { "BaseSerializerTypeName", "System.ComponentModel.Design.Serialization.DesignerSerializerAttributeBaseSerializerTypeName" };
+            yield return new object[] { "BaseSerializerTypeName,Other", "System.ComponentModel.Design.Serialization.DesignerSerializerAttributeBaseSerializerTypeName" };
+            yield return new object[] { string.Empty, "System.ComponentModel.Design.Serialization.DesignerSerializerAttribute" };
+            // .NET Framework throws NullReferenceException.
+            if (!PlatformDetection.IsFullFramework)
+            {
+                yield return new object[] { null, "System.ComponentModel.Design.Serialization.DesignerSerializerAttribute" };
+            }
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/tests/Design/Serialization/DesignerSerializerAttributeTests.cs]: Same question.  I don't see it in the Configuration.props.

File: src/System.ComponentModel.TypeConverter/tests/Design/Serialization/RootDesignerSerializerAttributeTests.cs [hunk 3]
```diff
@@ -55,29 +56,26 @@ public void Ctor_SerializerTypeName_BaseSerializerType(string serializerTypeName
             Assert.Equal(reloadable, attribute.Reloadable);
         }
 
+        public static IEnumerable<object[]> TypeId_TestData()
+        {
+            yield return new object[] { "BaseSerializerTypeName", "System.ComponentModel.Design.Serialization.RootDesignerSerializerAttributeBaseSerializerTypeName" };
+            yield return new object[] { "BaseSerializerTypeName,Other", "System.ComponentModel.Design.Serialization.RootDesignerSerializerAttributeBaseSerializerTypeName" };
+            yield return new object[] { string.Empty, "System.ComponentModel.Design.Serialization.RootDesignerSerializerAttribute" };
+            // .NET Framework throws NullReferenceException.
+            if (!PlatformDetection.IsFullFramework)
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/tests/Design/Serialization/RootDesignerSerializerAttributeTests.cs]: Same question.  Will stop asking.

File: src/System.ComponentModel.TypeConverter/tests/Design/MenuCommandTests.cs (hunk 1 above)
Comment (ViktorHofer) [reply src/System.ComponentModel.TypeConverter/tests/Design/MenuCommandTests.cs]: If the test project doesn't have a netfx configuration then not - in this case not.

Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/tests/Design/MenuCommandTests.cs]: So should I remove all the `PlatformDetection` stuff?

**Comment by hughbe** [comment]
@dotnet-bot test corefx-ci
@dotnet-bot test corefx-ci (Linux musl_x64_Debug)
@dotnet-bot test corefx-ci (Linux x64_Debug)

File: src/System.ComponentModel.TypeConverter/tests/Design/MenuCommandTests.cs (hunk 1 above)
Comment (ViktorHofer) [reply src/System.ComponentModel.TypeConverter/tests/Design/MenuCommandTests.cs]: All .NetFramework related code

**Comment by ViktorHofer** [comment]
And to just trigger the default pipeline without the outerloop legs you would say `azp run corefx-ci`.

**Comment by hughbe** [comment]
> System.Security.Cryptography.Rsa.Tests.EncryptDecrypt_Span.RsaDecryptAfterExport
Internal.Cryptography.CryptoThrowHelper+WindowsCryptographicException : Unknown error (0xc0000001)
   at System.Security.Cryptography.RSAImplementation.RSACng.TryEncryptOrDecrypt(SafeNCryptKeyHandle key, ReadOnlySpan`1 input, Span`1 output, AsymmetricPaddingMode paddingMode, Void* paddingInfo, Boolean encrypt, Int32& bytesWritten) in /_/src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs:line 266
   at System.Security.Cryptography.RSAImplementation.RSACng.TryEncryptOrDecrypt(ReadOnlySpan`1 data, Span`1 destination, RSAEncryptionPadding padding, Boolean encrypt, Int32& bytesWritten) in /_/src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs:line 205
   at System.Security.Cryptography.RSAImplementation.RSACng.TryDecrypt(ReadOnlySpan`1 data, Span`1 destination, RSAEncryptionPadding padding, Int32& bytesWritten) in /_/src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs:line 38
   at System.Security.Cryptography.Rsa.Tests.EncryptDecrypt_Span.<>c__DisplayClass1_0.<Decrypt>b__0(Byte[] dest) in /_/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.netcoreapp.cs:line 16
   at System.Security.Cryptography.Rsa.Tests.EncryptDecrypt_Span.TryWithOutputArray(Func`2 func) in /_/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.netcoreapp.cs:line 23
   at System.Security.Cryptography.Rsa.Tests.EncryptDecrypt_Span.Decrypt(RSA rsa, Byte[] data, RSAEncryptionPadding padding) in /_/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.netcoreapp.cs:line 16
   at System.Security.Cryptography.Rsa.Tests.EncryptDecrypt.RsaDecryptAfterExport() in /_/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.cs:line 615

**Comment by hughbe** [comment]
azp run corefx-ci

**Comment by hughbe** [comment]
Happy with this PR @stephentoub ? :)

**Comment by stephentoub** [comment]
We should hold off on taking this until after we branch for 3.0.  It's not necessary for the release.  Thanks.

**Comment by maryamariyan** [comment]
This PR seems to be ready now that we have master branch open for 5.0 submissions.

**Comment by danmosemsft** [comment]
It should be enough for you to sign off @maryamariyan

**Pull request merged**